script/gn-asar.js

```diff
@@ -46,13 +46,15 @@ try {
   }
 } catch (err) {
   console.error('Unexpected error while generating ASAR', err)
-  fs.removeSync(tmpPath)
-  process.exit(1)
+  fs.remove(tmpPath)
+    .then(() => process.exit(1))
+    .catch(() => process.exit(1))
+  return
 }
 
 // Create the ASAR archive
 asar.createPackageWithOptions(tmpPath, out[0], {}, (err) => {
-  fs.removeSync(tmpPath)
+  fs.remove(tmpPath)
 
   if (err) {
```

Comment:
Since fs.remove is async, it seems like it would be a good idea to:
a. check for error first
b. wait for fs.remove to finish before checking for error.

Reply:
Good point 👍